python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -173,20 +403,73 @@ def get_query_status(context, query_id):
             from snowflake.sqlalchemy import URL
             from sqlalchemy import create_engine
 
-            engine = create_engine(URL(**self.conn_args), connect_args=self.sqlalchemy_engine_args)
+            engine = create_engine(
+                URL(**self._sqlalchemy_connection_args), connect_args=self._sqlalchemy_engine_args
+            )
             conn = engine.raw_connection() if raw_conn else engine.connect()
 
             yield conn
             conn.close()
             engine.dispose()
         else:
-            conn = snowflake.connector.connect(**self.conn_args)
+            conn = snowflake.connector.connect(**self._connection_args)
 
             yield conn
             if not self.autocommit:
                 conn.commit()
             conn.close()
 
+    def get_object_to_set_on_execution_context(self) -> Any:
+        return SnowflakeConnection(
+            config=self._resolved_config_dict,
+            log=get_dagster_logger(),
+            snowflake_connection_resource=self,
```

Comment:
Just to make sure - `SnowflakeConnection` here is only used for backcompat reasons, e.g. for the old resource and for injecting the old resource into old-style ops?

Reply:
yes!

Reply:
can we leave a comment about the backcompat here?